### ISSUE TITLE
HTTP API: POST to exec - cmd not running as array, and erroring as string

### ISSUE THREAD

**Issue by lylebrown** [containers/podman#18289]
### Issue Description

I am attempting to run commands inside my podman containers via the HTTP API. In my test case, I am trying to stop the caddy server within a container named "caddy" using the command "caddy stop" which I can test by trying to access resources it reverse proxies. If I pass a single string as the `Cmd`, I get the following error suggesting it needs to be an array:

>  "cause": "json: cannot unmarshal string into Go struct field ExecCreateConfig.Cmd of type []string",
    "message": "decoding request body as JSON: json: cannot unmarshal string into Go struct field ExecCreateConfig.Cmd of type []string",
    "response": 500

This is strange, as the api docs state you can use a single string or an array.

If I replace the single string with an array containing only one string (`["caddy stop"]`), I get an Id as a response, but the command seemingly did not execute.

### Steps to reproduce the issue

Steps to reproduce the issue
1. Start a container called `caddy`
1. Start the podman API with `podman system service -t 0 tcp:ipaddress:port`
2. POST to http://`ipaddress`:`port`/v4.5.0/libpod/containers/caddy/exec with the following body:
```
{
    "Cmd": ["caddy stop"]
}
```

### Describe the results you received

Got a response with an Id, but command does not appear to have been executed in container.

### Describe the results you expected

Command should have been executed in the container (stopping the caddy service, in this instance).

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.30.0
  cgroupControllers:
  - cpuset
  - cpu
  - io
  - memory
  - hugetlb
  - pids
  - misc
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.7-2.fc37.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.7, commit: '
  cpuUtilization:
    idlePercent: 99.52
    systemPercent: 0.16
    userPercent: 0.33
  cpus: 24
  databaseBackend: boltdb
  distribution:
    distribution: fedora
    variant: server
    version: "37"
  eventLogger: journald
  hostname: brownsrv01
  idMappings:
    gidmap: null
    uidmap: null
  kernel: 6.1.9-200.fc37.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 22269939712
  memTotal: 134990086144
  networkBackend: netavark
  ociRuntime:
    name: crun
    package: crun-1.7.2-3.fc37.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.7.2
      commit: 0356bf4aff9a133d655dc13b1d9ac9424706cac4
      rundir: /run/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +LIBKRUN +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: false
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-8.fc37.x86_64
    version: |-
      slirp4netns version 1.2.0
      commit: 656041d45cfca7a4176f6b7eed9e4fe6c11e8383
      libslirp: 4.7.0
      SLIRP_CONFIG_VERSION_MAX: 4
      libseccomp: 2.5.3
  swapFree: 8448897024
  swapTotal: 8589930496
  uptime: 736h 1m 0.00s (Approximately 30.67 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
  - codeberg.org
  - ghcr.io
store:
  configFile: /usr/share/containers/storage.conf
  containerStore:
    number: 28
    paused: 0
    running: 15
    stopped: 13
  graphDriverName: overlay
  graphOptions:
    overlay.mountopt: nodev,metacopy=on
  graphRoot: /var/lib/containers/storage
  graphRootAllocated: 214643507200
  graphRootUsed: 45004300288
  graphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "true"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 52
  runRoot: /run/containers/storage
  transientStore: false
  volumePath: /var/lib/containers/storage/volumes
version:
  APIVersion: 4.5.0
  Built: 1681486976
  BuiltTime: Fri Apr 14 08:42:56 2023
  GitCommit: ""
  GoVersion: go1.19.7
  Os: linux
  OsArch: linux/amd64
  Version: 4.5.0
```

### Podman in a container

No

### Privileged Or Rootless

Privileged

### Upstream Latest Release

Yes

### Additional environment details

Running podman on Fedora Server on a physical server.

### Additional information

I have also tried including 
```
    "User": "root",
    "Privileged": true
```
in the POST request body, with the same results.

I have also tried separating the Cmd in the array per someone's suggestion `["caddy", "stop"]` and still get the same result.

**Comment by Luap99** [comment]
You have to start the exec session with the ID that was returned, https://docs.podman.io/en/latest/_static/api.html#tag/exec/operation/ExecStartLibpod

**Comment by lylebrown** [comment]
That did it, thank you! For reference, to anyone else who may stumble upon this, I did have to split the command into an array, so I used `"Cmd": ["caddy","stop"]` in the request body.

I'm not sure if this process of creating an exec session and then starting it with a separate request is common practice, but I feel like the documentation could use an additional comment about this. Even just a single sentence in the API docs in the exec section.

Feel free to close this if you want to track documentation change requests elsewhere (or if you believe it isn't needed) - my issue is resolved.
